mhs/common/mhs_common/workflow/tests/test_synchronous.py

```diff
@@ -0,0 +1,236 @@
+                                                                                       payload="nice message",
+                                                                                       work_description_object=None)
+
+        wdo.set_outbound_status.assert_called_with(work_description.MessageStatus.OUTBOUND_MESSAGE_PREPARATION_FAILED)
+        self.assertEqual(error, 500)
+        self.assertEqual(text, 'Error obtaining outbound URL')
+        log_mock.error.assert_called_with('009', 'Failed to retrieve details from spine route lookup')
+
+    @mock.patch.object(sync, 'logger')
+    @mock.patch('mhs_common.state.work_description.create_new_work_description')
+    @async_test
+    async def test_prepare_message_failure(self, wd_mock, log_mock):
+        wdo = mock.MagicMock()
+        wdo.publish.return_value = test_utilities.awaitable(None)
+        wd_mock.return_value = wdo
+        self.wf._lookup_to_asid_details = mock.MagicMock()
+        self.wf._lookup_to_asid_details.return_value = test_utilities.awaitable(("yes", "313123"))
+        self.wf._prepare_outbound_message = mock.MagicMock()
+        self.wf._prepare_outbound_message.side_effect = Exception()
+        wdo.set_outbound_status.return_value = test_utilities.awaitable(None)
+
+        error, text, work_description_response = await self.wf.handle_outbound_message(from_asid="202020",
+                                                                                       message_id="123",
+                                                                                       correlation_id="qwe",
+                                                                                       interaction_details={},
+                                                                                       payload="nice message",
+                                                                                       work_description_object=None)
+
+        wdo.set_outbound_status.assert_called_with(work_description.MessageStatus.OUTBOUND_MESSAGE_PREPARATION_FAILED)
+        self.assertEqual(error, 500)
+        self.assertEqual(text, 'Failed message preparation')
+        log_mock.error.assert_called_with('002', 'Failed to prepare outbound message')
+
+    @mock.patch('mhs_common.state.work_description.create_new_work_description')
+    @async_test
+    async def test_send_message_failure(self, wd_mock):
+        wdo = mock.MagicMock()
+        wdo.publish.return_value = test_utilities.awaitable(None)
+        wd_mock.return_value = wdo
+        self.wf._lookup_to_asid_details = mock.MagicMock()
+        self.wf._lookup_to_asid_details.return_value = test_utilities.awaitable(("yes", "313123"))
+        self.wf._prepare_outbound_message = mock.MagicMock()
+        self.wf._prepare_outbound_message.return_value = test_utilities.awaitable(("123", {"qwe": "qwe"}, "message"))
+        wdo.set_outbound_status.return_value = test_utilities.awaitable(None)
```

Comment:
Don't you need to make `self.transmission.make_request.return_value` raise an exception here?

Reply:
I think this passes because there's no async response on `make_request`, hence it errors anyway, but I'll add the exception anyway for completeness